spark/README.md

```diff
@@ -0,0 +1,109 @@
+# ESCAPE data science summer school 2021: Apache Spark
+
+Welcome to the Apache Spark lecture! Please, read carefuly this page before starting the lecture as it contains important information to set up the materials. 
+
+### Goals of the lecture
+
+*  Learning the basics of cluster computing, and the big data challenges in modern science.
+- Learning how to integrate scientific tools into [Apache Spark](https://spark.apache.org/), and how to perform efficient analysis on large volumes of data.
+
+### Timetable
+
+- **Session 1:** 
+  - I will review the landscape of cluster computing by addressing some of the most pressing questions today: what is cluster computing? What does it mean working in a distributed environment? What are the data and computing challenges that the scientific community is facing nowadays, and how can we tackle those? Some useful concepts like functional programming and implicit parallelisation will be discussed. I will also introduce Apache Spark, a cluster computing framework for analysing large datasets that proved successful in the industry. I will specifically focus on the Apache Spark SQL module and DataFrames API, and we will start practicing through a series of simple exercises.
+- **Session 2:** 
+  - In this session, we will use the Apache Spark Python API (PySpark) and learn on concrete examples how to interface and play with popular scientific libraries (Numpy, Pandas, ...). We will also see how to test and debug a code written with Spark, and integrate it in a Continuous Integration pipeline.
+- **Session 3:** 
+  - For the last session, we will finish with concrete applications in the domain of astronomy: catalog & image manipulation, machine learning and streaming data (if time permits).
+
+### Pre-requisites:
+
+- Latest version of [Docker](https://docs.docker.com/get-docker/) installed. In case you do not have it, all the exercises will be also accessible from Google Colab - but you will need a Google account.
+- A good knowledge in Python
+- Some general knowledge on parallel computing, and its challenges
+- Some knowledge in functional programming can help.
+
+
+### How to play the exercices?
+
+As Spark needs a specific environment, the best way is to use Docker. We provide a Dockerfile in the repo to build the image. From the `spark/` folder of the school repository, execute the build script:
+
+```bash
+./build_image.sh
+```
+
+The first time, it will download the `jupyter/pyspark-notebook:spark-3.1.1` image (~3.5GB), and install the necessary dependencies. It will also perform a test of the installation by printing some information at the end:
```

Comment:
I was surprised by the large size of 3.5 GB and checked dockerhub, the image is large, but "only" 1.2 GB: https://hub.docker.com/layers/jupyter/pyspark-notebook/spark-3.1.2/images/sha256-64853bb4e33b935bead2d616010e59e958a408bb35a4840bc2944552046b3a45?context=explore

Reply:
1.2GB is the compressed size.

Reply:
Text has been clarified https://github.com/escape2020/school2021/pull/55/commits/160b24c8b9ac45c53c17530210edc69a7b174b65